### PR TITLE
frontend: mock i18n.language on amount.test

### PR DESCRIPTION
**Pull request by shonsirsha**

### CODE REVIEW

File: frontends/web/src/components/amount/amount.test.tsx [hunk 1]
```diff
@@ -20,16 +20,20 @@ import { render } from '@testing-library/react';
...
   createContext: vi.fn()
 }));
 
+vi.mock('../../i18n/i18n', () => ({
+  i18n: {
+    language: 'de-CH',
+  },
+}));
+
+
```
Comment (shonsirsha) [inline frontends/web/src/components/amount/amount.test.tsx:36]: Later on, we can use `vi.mocked(i18n).language = 'en-US'` (or any locale) in different test cases to mock different locales. 

I don't think we've ever used `vi.mocked()` before in this project - but this is a way to mock a property of `i18n` module / object, in this case `language` (string).

Comment (thisconnect) [reply frontends/web/src/components/amount/amount.test.tsx:36]: thank you so much

**Comment by thisconnect** [comment]
thanks I cherry picked your commit, now trying to mock different valus for each test but I got stuck again :(

seems as en-US is overwriting de-CH 🤔 

https://github.com/thisconnect/bitbox-wallet-app/tree/frontend-amount-refactor

could you give it a shot?

**Comment by thisconnect** [comment]
closing this

**Comment by thisconnect** [comment]
maybe somewhat similar problem https://github.com/vitest-dev/vitest/discussions/4124
?

**Comment by thisconnect** [comment]
I guess a good alternative could be that Amount could just have a language prop instead of importing i18n.

**Pull request closed without merging**